Sources/Testing/Running/EntryPoint.swift

```diff
@@ -0,0 +1,82 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+@_implementationOnly import TestingInternals
+
+#if !SWT_TARGET_OS_APPLE
+/// The entry point to the testing library used by Swift Package Manager.
+///
+/// - Returns: The result of invoking the testing library. The type of this
+///   value is subject to change.
+///
+/// This function examines the command-line arguments to the current process
+/// and then invokes available tests in the current process.
+///
+/// - Warning: This function is used by Swift Package Manager. Do not call it
+///   directly.
+@_spi(SwiftPackageManagerSupport)
+public func __swiftPMEntryPoint() async -> CInt {
+  let args = CommandLine.arguments()
+  // We do not use --dump-tests-json to handle test list requests. If that
+  // argument is passed, just exit early.
+  if args.contains("--dump-tests-json") {
+    return EXIT_SUCCESS
+  }
+
+  @Locked var exitCode = EXIT_SUCCESS
+  await runTests { event, _ in
+    if case let .issueRecorded(issue) = event.kind, !issue.isKnown {
+      $exitCode.withLock { exitCode in
+        exitCode = EXIT_FAILURE
+      }
+    }
+  }
+  return exitCode
+}
+#endif
+
+/// The common implementation of `__main()` and
```

Comment:
```suggestion
/// The common implementation of `__swiftPMEntryPoint()` and
```
or whatever final name you go with